app/api/localStorage/primitives.js

```diff
@@ -14,80 +14,143 @@ declare var chrome;
 //  get
 // =====
 
-const getStorageItemInExtension = (key: ?string) => new Promise((resolve, reject) => {
-  chrome.storage.local.get(key, (data, error) => {
+const getStorageItemInExtension = async (
+  key: string | void
+): Promise<?string> => new Promise((resolve, reject) => {
+  chrome.storage.local.get(key, (data: {}, error) => {
     if (error) reject(error);
-    if (!key) resolve(data);
-    const value = data[key];
-    if (value === undefined) resolve('');
-    resolve(value);
+    if (key === undefined) {
+      resolve(JSON.stringify(data));
+    } else {
+      const value: any = data[key];
+      // need to ensure type is string to match localStorage API
+      if (value == null) {
+        resolve(value);
+      } else if (typeof value === 'string') {
+        resolve(value);
+      } else {
+        reject(new Error('getStorageItemInExtension cannot get non-string value'));
+      }
+    }
   });
 });
 
-const getStorageItemInWeb = (key: ?string) => new Promise((resolve, reject) => {
-  try {
-    if (!key) return resolve(JSON.stringify(localStorage));
-    const value = localStorage.getItem(key);
-    if (!value) return resolve('');
-    resolve(value);
-  } catch (error) {
-    return reject(error);
-  }
-});
+const getStorageItemInWeb = async (
+  key: string | void
```

Comment:
why sometimes we use `string | void` and other times `?string`?

Reply:
`string | void`  doesn't allow `null` but `?string` does